### PR TITLE
Gen5 PAX code change

### PR DESCRIPTION
**Pull request by xyan264**
As Gen5 PAX hardware is near, Switchtec-user needs updates to support Gen5 PAX:
- remove unnecessary Gen4 checking in PAX (as Gen5 PAX is available soon)
- use new MRPC subcommand ID, as defined by Gen5 PAX hardware specification
- accommodate for the data structure changes found in in Gen5 PAX

### CODE REVIEW

**Review by lsgunth — APPROVED** [review]
Looks good to me, thanks.

File: lib/fabric.c [hunk 1]
```diff
@@ -121,25 +123,86 @@ enum switchtec_fab_topo_info_dump_status {
...
+
+	ret = topo_info_dump_start(dev);
+	if(ret)
+		return ret;
+
+	do {
+		ret = topo_info_dump_status_get(dev, &status, &total_info_len);
+		if(ret)
+			return ret;
+	} while(status == SWITCHTEC_FAB_TOPO_INFO_DUMP_WAIT);
```
Comment (kelvin-cao) [inline lib/fabric.c:152]: Style nit: a white space is required between 'while' and '('

File: lib/fabric.c [hunk 2]
```diff
@@ -121,25 +123,86 @@ enum switchtec_fab_topo_info_dump_status {
...
 
-	if (!switchtec_is_gen4(dev) || !switchtec_is_pax_all(dev)) {
-		errno = ENOTSUP;
+		struct switchtec_fab_port_info list[SWITCHTEC_MAX_PORTS];
+	} reply = {};
+
+	char *buf = (char *)&reply;
+
+	ret = topo_info_dump_start(dev);
+	if(ret)
```
Comment (kelvin-cao) [inline lib/fabric.c:145]: Style nit: white space required between 'if' and '('

File: lib/fabric.c [hunk 3]
```diff
@@ -121,25 +123,86 @@ enum switchtec_fab_topo_info_dump_status {
...
+		return ret;
+
+	do {
+		ret = topo_info_dump_status_get(dev, &status, &total_info_len);
+		if(ret)
```
Comment (kelvin-cao) [inline lib/fabric.c:150]: Ditto

File: lib/fabric.c [hunk 4]
```diff
@@ -121,25 +123,86 @@ enum switchtec_fab_topo_info_dump_status {
...
+	if (total_info_len > sizeof(reply))
 		return -1;
+
+	offset = 0;
+	buf_len = sizeof(reply);
+
+	while (offset < total_info_len) {
+		ret = topo_info_dump_data_get(dev, offset,
+					      buf+offset, &buf_len);
+		if(ret)
```
Comment (kelvin-cao) [inline lib/fabric.c:166]: Ditto

File: lib/fabric.c [hunk 5]
```diff
@@ -166,10 +229,42 @@ int switchtec_topo_info_dump(struct switchtec_dev *dev,
...
+		buf_len = sizeof(reply) - offset;
+	}
+
+	ret = topo_info_dump_finish(dev);
+	if (ret)
+		return ret;
+
+	topo_info->sw_idx = reply.sw_idx;
+	topo_info->num_stack_bif = 8;
+	memcpy(topo_info->stack_bif, reply.stack_bif, 8*sizeof(uint32_t));
```
Comment (kelvin-cao) [inline lib/fabric.c:241]: Style nit: white spaces surround '*'.

File: lib/fabric.c [hunk 6]
```diff
@@ -166,10 +229,42 @@ int switchtec_topo_info_dump(struct switchtec_dev *dev,
...
+	memcpy(topo_info->route_port, reply.route_port, 16*sizeof(uint8_t));
```
Comment (kelvin-cao) [inline lib/fabric.c:242]: Ditto

**Review by kelvin-cao — APPROVED** [review]
Looks good except for some style nits (not all pointed out).

**Pull request merged**